### PR TITLE
MON-3701: remove references to OAuth proxy

### PR DESCRIPTION
**Pull request by simonpasquier**
* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Review by marioferh — COMMENTED** [review]
/lgtm

File: test/e2e/alertmanager_test.go [hunk 1]
```diff
@@ -482,8 +482,8 @@ func TestAlertmanagerDataReplication(t *testing.T) {
 	}
 }
 
-// The Alertmanager API should be protected by the OAuth proxy.
-func TestAlertmanagerOAuthProxy(t *testing.T) {
+// The Alertmanager API should be protected by authentication/authorization.
+func TestAlertmanagerAPI(t *testing.T) {
```
Comment (machine424) [inline test/e2e/alertmanager_test.go:486]: I don't see how the test is checking the auth

**Review by machine424 — COMMENTED** [review]
Thanks.
I see some other references that we can get rid of as well:
- https://github.com/openshift/cluster-monitoring-operator/blob/7f498b49343756681f34d0d75013f193cfdc8a66/jsonnet/components/prometheus-user-workload.libsonnet#L57
- https://github.com/openshift/cluster-monitoring-operator/blob/7f498b49343756681f34d0d75013f193cfdc8a66/examples/user-guides/configuring-cluster-monitoring/custom-image-config.yaml#L10 (I'm not even sure if we still consume that)
- https://github.com/openshift/cluster-monitoring-operator/blob/7f498b49343756681f34d0d75013f193cfdc8a66/test/e2e/thanos_querier_test.go#L75 (I don't see what that test is checking, maybe we should improve the utils)

File: test/e2e/alertmanager_test.go (hunk 1 above)
Comment (machine424) [reply test/e2e/alertmanager_test.go:486]: (not related to the PR)

Comment (machine424) [reply test/e2e/alertmanager_test.go:486]: cc @simonpasquier if you missed this (I forgot o hold)
I could take care of it if you want.

**Comment by simonpasquier** [comment]
@machine424 

> I see some other references that we can get rid of as well:

Yes I plan to submit separate PRs to smooth the reviews and CI debugging.

> I don't see how the test is checking the auth

It's not very explicit but the test SA has the necessary RBAC to query the API. Testing that unauthorized users are rejected would be good indeed (we have it for tenancy services but for not admin-like services).

**Pull request merged**